ChatAPI.Application/UseCases/Implementations/FriendshipService.cs

```diff
@@ -47,7 +65,23 @@ public async Task<Result> AddFriend(int senderUserId, string phone)
 		/// <param name="senderUserId">The id of the user who sent the friend request.</param>
 		/// <param name="targetUserId">The targeted user id (receiver).</param>
 		/// <param name="newStatus">The status to update with.</param>
-		public Task RespondToFriendRequest(int senderUserId, int targetUserId, FriendshipStatus newStatus) =>
-			friendRepo.UpdateFriendshipStatus(senderUserId, targetUserId, newStatus);
+		public async Task<Result> RespondToFriendRequest(int senderUserId, int targetUserId, FriendshipStatus newStatus)
+		{
+			Friendship? friendship = await friendRepo.GetSpecificFriendshipByUserIds(senderUserId, targetUserId);
+
+			if (friendship is null)
+				return Result.Failure("Could not find a friend request.", HttpStatusCode.NotFound);
+
+			if (newStatus is FriendshipStatus.Pending)
+				return Result.Failure("You cannot change the status to pending.", HttpStatusCode.Forbidden);
+
+			/******** [CHAT-23] History of Friendship updates ********/
+
+			//if (friendship.Status is FriendshipStatus.Blocked or FriendshipStatus.Rejected &&
+			//	friendship.ModifiedBy == senderUserId)
+			//	return Result.Failure("You cannot change the status.", HttpStatusCode.Forbidden);
+
+			return Result.Success();
+		}
```

Comment:
This method is supposed to accept or decline a friend request. It does no such thing.